### PR TITLE
Added a boolean to the email rest model

### PR DESCRIPTION
**Pull request by bamandel**
Added a boolean field which will let us know whether the password field is set or not similar to how we did it previously.

### CODE REVIEW

File: service-email/src/main/java/com/synopsys/integration/alert/service/email/model/EmailGlobalConfigModel.java [hunk 1]
```diff
@@ -41,8 +43,8 @@ public EmailGlobalConfigModel obfuscate() {
         emailGlobalConfigModel.setSmtpUsername(smtpUsername);
         emailGlobalConfigModel.setAdditionalJavaMailProperties(additionalJavaMailProperties);
 
-        String maskedPassword = (smtpPassword != null) ? AlertRestConstants.MASKED_VALUE : null;
-        emailGlobalConfigModel.setSmtpPassword(maskedPassword);
+        emailGlobalConfigModel.setIsSmtpPasswordSet(StringUtils.isNotBlank(smtpPassword));
+        emailGlobalConfigModel.setSmtpPassword(null);
```
Comment (ChomickiM) [inline service-email/src/main/java/com/synopsys/integration/alert/service/email/model/EmailGlobalConfigModel.java:47]: Is the idea to remove (or perhaps rename?) the obfuscate method that is running this? If so, we would probably need to also refactor the ConfigurationCrudHelper and the Obfuscated interface

Comment (bamandel) [reply service-email/src/main/java/com/synopsys/integration/alert/service/email/model/EmailGlobalConfigModel.java:47]: Well it's still obfuscating by removing the sensitive data entirely i'd argue. We could change the name to something clearer

**Review by ChomickiM — APPROVED** [review]

File: channel-email/src/main/java/com/synopsys/integration/alert/channel/email/database/accessor/EmailGlobalConfigAccessor.java [hunk 2]
```diff
@@ -132,8 +136,10 @@ private EmailGlobalConfigModel createConfigModel(EmailConfigurationEntity emailC
             newModel.setSmtpFrom(emailConfiguration.getSmtpFrom());
             newModel.setSmtpAuth(emailConfiguration.getAuthRequired());
             newModel.setSmtpUsername(emailConfiguration.getAuthUsername());
-            if (StringUtils.isNotBlank(emailConfiguration.getAuthPassword())) {
-                newModel.setSmtpPassword(encryptionUtility.decrypt(emailConfiguration.getAuthPassword()));
+            String authPassword = emailConfiguration.getAuthPassword();
+            newModel.setIsSmtpPasswordSet(StringUtils.isNotBlank(authPassword));
+            if (StringUtils.isNotBlank(authPassword)) {
+                newModel.setSmtpPassword(encryptionUtility.decrypt(authPassword));
```
Comment (psantos1113) [inline channel-email/src/main/java/com/synopsys/integration/alert/channel/email/database/accessor/EmailGlobalConfigAccessor.java:142]:
```suggestion
            String authPassword = emailConfiguration.getAuthPassword();
  boolean passwordSetInDB = StringUtils.isNotBlank(authPassword);
  newModel.setIsSmtpPasswordSet(passwordSetInDB);
  if(passwordSetInDB) {
      newModel.setSmtpPassword(encryptionUtility.decrypt(authPassword));
```

File: channel-email/src/main/java/com/synopsys/integration/alert/channel/email/database/accessor/EmailGlobalConfigAccessor.java [hunk 3]
```diff
@@ -93,6 +94,9 @@ public EmailGlobalConfigModel createConfiguration(EmailGlobalConfigModel configu
     public EmailGlobalConfigModel updateConfiguration(UUID configurationId, EmailGlobalConfigModel configuration) throws AlertConfigurationException {
         EmailConfigurationEntity configurationEntity = emailConfigurationRepository.findById(configurationId)
             .orElseThrow(() -> new AlertConfigurationException(String.format("Config with id '%s' did not exist", configurationId.toString())));
+        if (BooleanUtils.toBoolean(configuration.getIsSmtpPasswordSet())) {
+            configuration.setSmtpPassword(configurationEntity.getAuthPassword());
+        }
```
Comment (bamandel) [inline channel-email/src/main/java/com/synopsys/integration/alert/channel/email/database/accessor/EmailGlobalConfigAccessor.java:99]: Need to add check to see if password exists in new body, use that new value instead

**Review by psantos1113 — APPROVED** [review]
Just a small cleanup suggestion but otherwise good.

**Pull request merged**